scripts/reboot_helper.py

```diff
@@ -0,0 +1,224 @@
+    load_platform_chassis()
+
+    if not is_smartswitch():
+        log.log_error("Platform is not a smartswitch to reboot module")
+        return False
+
+    # Attempt to reboot the module
+    if not hasattr(platform_chassis, 'reboot'):
+        log.log_error("Reboot method not found in platform chassis")
+        return False
+
+    if module_name.upper() not in get_all_dpus():
+        log.log_error("Module {} not found".format(module_name))
+        return False
+
+    log.log_info("Rebooting module {}...".format(module_name))
+    try:
+        platform_chassis.reboot(module_name)
+        log.log_info("Reboot command sent for module {}".format(module_name))
+    except NotImplementedError:
+        raise NotImplementedError("Reboot not implemented on this platform: {type(e).__name__}")
+    except Exception as e:
+        log.log_error("Unexpected error occurred while rebooting module {}: {}".format(module_name, e))
+        return False
+
+    return True
+
+
+def is_dpu():
+    """Check if script is running on DPU module"""
+
+    # Load the platform chassis if not already loaded
+    load_platform_chassis()
+
+    if not is_smartswitch():
+        return False
+
+    # Load platform.json
+    platform_info = device_info.get_platform_info()
+    platform = platform_info.get('platform')
+    if not platform:
+        log.log_error("Platform does not exist in platform_info")
+        return False
+
+    config_data = load_platform_json(platform)
+    if config_data is None:
+        return False
+
+    return any(re.search(r'\.DPU$', key) for key in config_data.keys())
+
+
+def pci_detach_module(module_name):
+    """
+    Detach the specified module by invoking the platform API.
+
+    Args:
+        module_name (str): The name of the module to detach.
+
+    Returns:
+        bool: True if the detach command was successfully sent, False otherwise.
+    """
+
+    # Load the platform chassis if not already loaded
+    load_platform_chassis()
+
+    if not is_smartswitch():
+        log.log_error("Platform is not a smartswitch to detach module")
+        return False
+
+    # Attempt to detach the module
+    if not hasattr(platform_chassis, 'pci_detach'):
+        log.log_error("PCI detach method not found in platform chassis")
+        return False
+
+    if module_name.upper() not in get_all_dpus():
+        log.log_error("Module {} not found".format(module_name))
+        return False
+
+    log.log_info("Detaching module {}...".format(module_name))
+    try:
+        platform_chassis.pci_detach(module_name)
+        log.log_info("Detach command sent for module {}".format(module_name))
+        return True
+    except NotImplementedError:
+        raise NotImplementedError("PCI detach not implemented on this platform: {type(e).__name__}")
+    except Exception as e:
+        log.log_error("Unexpected error occurred while detaching module {}: {}".format(module_name, e))
+        return False
+
+
+def main():
```

Comment:
@vvolam main() function in helper?

Reply:
@prgeor This is to write unit tests for main function as well for maximum code coverage, else I couldn't call in unit tests.